ci/build_docs_pre_install.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+# Copyright (c) 2024 NVIDIA CORPORATION.
+#
+# [description]
+#
+#   ucx-py's docs builds require installing the library.
+#
+#   It does that by running 'pip install .' from the root of the repo. This script
+#   is used to modify readthedocs' local checkout of this project's source code prior
+#   to that 'pip install' being run.
+#
+#   For more, see https://docs.readthedocs.io/en/stable/build-customization.html
+#
+
+set -euo pipefail
+
+sed -r -i "s/libucx==(.*)\"/libucx-cu12==\1\"/g" ./pyproject.toml
```

Comment:
I'm proposing bringing back this script for docs builds, but having it substitute in a real, suffixed name for `libucx-cu12` so that docs builds will really install that.

I think that's necessary to meet these 3 constraints:

* docs builds run `pip install .` to install `ucx-py`
* fallback matrices in `dependencies.yaml` need to be the unsuffixed names (e.g. `libucx` not `libucx-cu12`)
   - *per the offline discussion that came out of https://github.com/rapidsai/rapids-build-backend/pull/29#issuecomment-2103461693*
* `pre-commit` hook running `rapids-dependency-file-generator` is going to put into `pyproject.toml` whatever is in the fallback matrices in `dependencies.yaml`